### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by Josphat205**
### **IN THIS BRANCH**

- Splash screen

- Added a simple page with the name of your app 

- The user should be able to register in the app with full name, email and password 
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in

- When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
  - Transactions page For a given category, the list of transactions is presented, ordered by the most recent
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

- Added a new transaction page
-  AddedTesting of model and controller

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,200 @@
...
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project you can support me by reaching me through my social media links above
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank microverse and my group members for the support they gave me during project development
```
Comment (MeqdamAlqudah) [inline README.md]: - [ ] Please the README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/views/shared/_splash.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class="container-fluid d-flex flex-column align-items-center justify-content-center splash-screen">
+    <h1 class="text-white">Budget App 🪙</h1>
+    <br/>
+      <div class="d-flex">
```
Comment (MeqdamAlqudah) [inline app/views/shared/_splash.html.erb]: - [ ] Please the Splash screen should contain the name of the project , I can see that you add the title but for some reason the title color is white , fix it please and make sure it is visible to the user 
image

File: app/models/expense.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Expense < ApplicationRecord
+  belongs_to :user
+  belongs_to :group
```
Comment (MeqdamAlqudah) [inline app/models/expense.rb]: - [x] Please there should be many to many relation between the Expense and the Group, models as in the database schema as I can see you create a one to many relation
image

File: app/views/groups/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<div class="container-fluid w-100 d-flex flex-column align-items-center mt-5">
+    <h1>Create a new category:</h1>
+    <%= form_with model: @group, url: groups_path, method: :post, :html => {:class => "w-75 mt-4"} do |form| %>
+      <div class="form-floating mb-3 w-100">
+        <%= form.text_field :name, class:"form-control", id:"floatingInput", placeholder:"Category name" %>
+        <%= form.label :name, 'Name', class: "form-label" %>
+      </div>
+      <div class="form-floating mb-3 w-100">
+        <%= form.text_field :icon, class:"form-control", id:"floatingInput", placeholder:"Category icon"  %>
+        <%= form.label :icon, 'Icon link', class: "form-label" %>
```
Comment (MeqdamAlqudah) [inline app/views/groups/new.html.erb]: - [ ] Please make sure that add a category and Add a new transaction design matches the one in the original design
image
image
The design 
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @lomantech , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## Highlights
:+1: categories blocks
:+1: files structure

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/expense.rb (hunk 3 above)
Comment (Josphat205) [reply app/models/expense.rb]: Added

File: app/views/shared/_splash.html.erb (hunk 2 above)
Comment (HAMMAS-SALEEM) [reply app/views/shared/_splash.html.erb]: 👍

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey @Josphat205 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

## Mandatory Requirements ♻️

- [ ] It would be great if you configure stylelint in your linters setup because you have some code of CSS to check its errors you should have stylelint configured in your project 🤔

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Josphat205, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well fixing the required changes and adding the linters for the CSS.

## Optional suggestions

- [ ] Let's always remember that there is a difference between GitHub flow and gitflow when we are setting up GitHub workflows. Your project was supposed to be invalidated in the first place.

git


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**